### PR TITLE
[MRG] FIX Handle cluster configuration whenever proxy are provided to launcher...

### PR DESCRIPTION
**Pull request by arjoly**
... / status job command

### CODE REVIEW

**Comment by arjoly** [comment]
@ogrisel This is ready for review. It appears that cluster distribution as rock roll cluster provides proxy for qstat / qsub command while you are on a slurm cluster. Instead of checking for the presence of those programs, it checks now if administrator tools are available. I expect that this will be more robust. At least, the tests are now passing on the new cluster.

File: clusterlib/scheduler.py [hunk 1]
```diff
@@ -53,10 +53,15 @@ def is_exe(fpath):
...
-        ('sge', 'qstat'),
-    )
+    # To detect if the backend is available, it's better to detect the presence
+    # of administrator tools over launching (sbatch, qsub) or job status
+    # (qstat, squeue) commands since a proxy might be provided by
+    # the cluster distribution such as the rock roll cluster distribution.
+    backend_commands = {
+        'slurm': 'scontrol',
+        'sge': 'qmod',
+    }
```
Comment (ogrisel) [inline clusterlib/scheduler.py]: The ordering of a Python dict is not deterministic. The use of a tuple of pairs as intended to keep the ordering of the tests deterministic to select slurm over sge when both are available on the host.

**Comment by ogrisel** [comment]
Other than my comment, LGTM.

File: clusterlib/scheduler.py (hunk 1 above)
Comment (arjoly) [inline clusterlib/scheduler.py]: Good point!

**Comment by arjoly** [comment]
Merged by rebase.

**Pull request merged**